apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -255,10 +261,19 @@ export function JsonFormsLinkControl({
   required,
 }: ControlProps) {
   const dataString = data && typeof data === "string" ? data : ""
+  // NOTE: We need to pass in `siteId` but this component is automaticall used by JsonForms
+  // so we are unable to pass props down
+  const { siteId } = useQueryParse(siteSchema)
+  const dummyFile = new File(
+    [],
+    // NOTE: Technically guaranteed since our s3 filepath has a format of `/<site>/.../<filename>`
+    dataString.split("/").at(-1) ?? "Uploaded file",
```

Comment:
nit: hmm we can't guarantee this right, because this covers normal links and files, so the link may not necessarily be a File.

But this is covered under the LinkHrefEditor so I'm guessing it should be fine?

Reply:
yeap we can't, but this is wrapped under the radio button as you've rightly pointed out so this case is technically impossible.

we could also use a regex this issue as the link itself adhere to strict formatting